binding/client.go

```diff
@@ -729,6 +759,12 @@ func (r *Client) send(rb func() (*http.Request, error)) (response *http.Response
 					response.StatusCode,
 					request.Method,
 					request.URL.Path))
+			if response.StatusCode == http.StatusGatewayTimeout {
```

Comment:
This looks little risky to me (no way out the loop when gateway timeout returned), but hopefully on some bigger failure, service unavailable or something else like that woud be returned.